### ISSUE TITLE
Allow to set global defaults for ApiResource attributes and other options

### ISSUE THREAD

**Issue by dunglas** [api-platform/core#3099]
It would be nice to configure global defaults for all class marked with `@ApiResource` (or configured in YAML/XML etc).

It could be a new config section such as the following:

```yaml
api_platform:
  defaults:
    attributes: # ...
    item_operations: # ...
    collection_operations: # ...
    normalization_context: # ...
    graphql: # ...
    # ...

# All properties defined in https://github.com/api-platform/core/blob/master/src/Annotation/ApiResource.php should be configurable
```

For instance, it would allow to make the API globally read only:

```yaml
api_platform:
    defaults:
        itemOperations: ["get"]
        collectionOperations: ["get"]
```

Other example, enable Mercure for all resource:

```yaml
api_platform:
    defaults:
        mercure: true
```

Or always use Messenger:

```yaml
api_platform:
    defaults:
        messenger: true
```

Implementing this feature should be trivial:

* add the new config option in https://github.com/api-platform/core/blob/master/src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php
* Add a new `ResourceMetadataFactory`, with a low priority, applying the defaults if not set

Most options are stored in metadata's attributes, some other are top level metadata properties (itemCollections and collectionOperations and graphql AFAIR). Basically, the content of `defaults` should be passed as-is to resources' attributes, except the special one previously mentioned.

Existing configuration option for resource attributes (pagination, cache headers...) should probably be deprecated for consistency.

**Comment by teohhanhui** [comment]
```yaml
api_platform:
    defaults:
        item_perations: ["get"]
        collection_perations: ["get"]
```

I don't think we need this. It'll make things very confusing...

**Comment by xavren** [comment]
Could be nice to configure defaults operations to protect by default all `@ApiResource` this could be very usefull i think

**Comment by julienfalque** [comment]
Is there anything left to do to implement this after #3151?

**Comment by soyuka** [comment]
Indeed this should be fixed by #3151

**Comment by teohhanhui** [comment]
I remember @dunglas discovered an issue with `cache_headers`.
